framework/Optimizers/Optimizer.py

```diff
@@ -620,229 +309,158 @@ def checkConvergence(self):
-      @ In, None
+      Stores information to the solution export.
+      @ In, traj, int, trajectory which should be written
+      @ In, rlz, dict, collected point
+      @ In, acceptable, bool, acceptability of opt point
       @ Out, None
     """
-    # README: this method is necessary because the base class optimizer doesn't know what needs to be reset in the
-    #         subclass, but the subclass doesn't know when it needs to call this method.
-    pass
 
-  @abc.abstractmethod
-  def _createEvaluationIdentifier(self, *args, **kwargs):
+  def _addTrackingInfo(self, info, **kwargs):
     """
-      Creates a unique ID to identifiy particular realizations as they return from the JobHandler.
-      Inputs should be specific to the needs of individual optimizers.
-      @ In, args, list, list of arguments
+      Creates realization identifiers to identifiy particular realizations as they return from the JobHandler.
+      Expandable by inheritors.
+      @ In, info, dict, dictionary of potentially-existing added identifiers
       @ In, kwargs, dict, dictionary of keyword arguments
-      @ Out, identifier, str, the evaluation identifier
+      @ Out, None (but "info" gets modified)
     """
-    pass
+    # TODO shouldn't this require the realization and information to do right?
+    info['traj'] = kwargs['traj']
 
-  def denormalizeData(self, optVars):
-    """
-      Method to normalize the data
-      @ In, optVars, dict, dictionary containing the value of decision variables to be deormalized, in form of {varName: varValue}
-      @ Out, optVarsDenorm, dict, dictionary containing the value of denormalized decision variables, in form of {varName: varValue}
-    """
-    optVarsDenorm = {}
-    for var in optVars.keys():
-      try:
-        optVarsDenorm[var] = optVars[var]*(self.optVarsInit['upperBound'][var]-self.optVarsInit['lowerBound'][var])+self.optVarsInit['lowerBound'][var]
-      except KeyError:
-        optVarsDenorm[var] = optVars[var]
-    return optVarsDenorm
-
-  def endJobRunnable(self):
-    """
-      Returns the maximum number of inputs allowed to be created by the optimizer right after a job ends
-      @ In, None
-      @ Out, endJobRunnable, int, number of runnable jobs at the end of each job
-    """
-    return self._endJobRunnable
-
-  def _expandVectorVariables(self):
+  def checkConstraint(self, optVars):
     """
-      Normally used to extend variables; in the Optimizer, we do that in localGenerateInput
-      @ In, None
-      @ Out, None
+      Method to check whether a set of decision variables satisfy the constraint or not in UNNORMALIZED input space
+      @ In, optVars, dict, dictionary containing the value of decision variables to be checked, in form of {varName: varValue}
+      @ Out, satisfaction, tuple, (bool,list) => (variable indicating the satisfaction of constraints at the point optVars, masks for the under/over violations)
     """
-    pass
+    TODO
```

Comment:
commented?

Reply:
this ended up being unused in the final design. Removed.